Automattic/jetpack
# Extract/jetpack plugin installer

This PR extracts the Jetpack_Plugins class into its own package.

In the future we might want to extract the REST API endpoints that use it to the package as well.

The approach was to keep the same class names and add the new package as a dependency to the Jetpack plugin. Doing that, the autoloader will take care to include the class whenever it's needed.

#### Changes proposed in this Pull Request:

* extracts the Jetpack_Plugins class into its own package.

#### Jetpack product discussion


p1643119059163700-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Set up a new Jurassic Ninja site with this branch
* Activate and connect Jetpack
* Add a new post
* Add a Contact Form block
* On the block options panel, go to "Newsletter Integration"
* Click to install the Creative Email plugin
* Make sure the plugin is properly installed and activated

The above tests the sideloading of plugins. Now let's test the REST endpoint that allows user to upload a zip file to add plugins to their site

* Go to wordpress.com
* Switch site to your testing Jetpack site
* Go to Plugins
* Click "Upload" on the top of the screen
* Upload a zip file with any valid plugin
* Go back to your site's wp-admin and verify that the plugin was installed and activated.

Now let's test the backward compatibility of the changes. Let's open a terminal and do things the old way.

Open a terminal with `wp shell`

* `jetpack_require_lib( 'plugins' );` should return null
* Install a plugin using the old class, for example woocommerce:
* `Jetpack_Plugins::install_plugin( 'woocommerce' );`
* Verify that the plugin is installed
